### ISSUE TITLE
[BUG] symbol __aeabi_memclr8@@LIBC_N has undefined version LIBC_N

### ISSUE THREAD

**Issue by mkzpd** [android/ndk#1789]
### Description

I want to build static link library of grpc on macOS,my macOS system is macOS Monterey version 12.6(Intel chip).

I found 

https://github.com/grpc/grpc/blob/master/BUILDING.md#building-with-cmake

That illustrates build shared libraries that is Dynamic-link library(.so file)，but I want to get static-link library.

So I install openssl with below command:
```
mindgood@minds-MacBook-Pro lib % brew install openssl
Warning: openssl@3 3.0.5 is already installed and up-to-date.
To reinstall 3.0.5, run:
  brew reinstall openssl@3
mindgood@minds-MacBook-Pro lib % openssl version
OpenSSL 3.0.5 5 Jul 2022 (Library: OpenSSL 3.0.5 5 Jul 2022)
mindgood@minds-MacBook-Pro lib % whereis opnessl
opnessl:
mindgood@minds-MacBook-Pro lib % find / -name 'openssl'
/usr/bin/openssl
find: /usr/sbin/authserver: Permission denied
/usr/local/ssl/bin/openssl
/usr/local/ssl/include/openssl
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Aliases/openssl
/usr/local/opt/openssl
/usr/local/Cellar/crystal/1.6.0/share/crystal/src/openssl
/usr/local/Cellar/openssl@3/3.0.5/bin/openssl
/usr/local/Cellar/openssl@3/3.0.5/include/openssl
/usr/local/Cellar/openssl@3/3.0.5/share/doc/openssl
/usr/local/Cellar/openssl@1.1/1.1.1q/bin/openssl
/usr/local/Cellar/openssl@1.1/1.1.1q/include/openssl
/usr/local/Cellar/openssl@1.1/1.1.1q/share/doc/openssl
find: /Library/Application Support/Apple/ParentalControls/Users: Permission denied
find: /Library/Application Support/Apple/AssetCache/Data: Permission denied
find: /Library/Application Support/ApplePushService: Permission denied
find: /Library/Application Support/com.apple.TCC: Operation not permitted
/Library/Developer/CommandLineTools/SDKs/MacOSX11.3.sdk/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/openssl
/Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/openssl
/Library/Developer/CommandLineTools/SDKs/MacOSX12.3.sdk/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/openssl
find: /Library/Caches/com.apple.iconservices.store: Permission denied
find: /Library/Caches/com.apple.aned: Operation not permitted
^C
mindgood@minds-MacBook-Pro grpc_csharp_ext.dir % brew info openssl
==> openssl@3: stable 3.0.5 (bottled) [keg-only]
Cryptography and SSL/TLS Toolkit
https://openssl.org/
/usr/local/Cellar/openssl@3/3.0.5 (6,444 files, 28.2MB)
  Poured from bottle on 2022-10-18 at 17:24:29
From: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/openssl@3.rb
License: Apache-2.0
==> Dependencies
Required: ca-certificates ✔
==> Caveats
A CA file has been bootstrapped using certificates from the system
keychain. To add additional certificates, place .pem files in
  /usr/local/etc/openssl@3/certs

and run
  /usr/local/opt/openssl@3/bin/c_rehash

openssl@3 is keg-only, which means it was not symlinked into /usr/local,
because macOS provides LibreSSL.

If you need to have openssl@3 first in your PATH, run:
  echo 'export PATH="/usr/local/opt/openssl@3/bin:$PATH"' >> ~/.zshrc

For compilers to find openssl@3 you may need to set:
  export LDFLAGS="-L/usr/local/opt/openssl@3/lib"
  export CPPFLAGS="-I/usr/local/opt/openssl@3/include"

For pkg-config to find openssl@3 you may need to set:
  export PKG_CONFIG_PATH="/usr/local/opt/openssl@3/lib/pkgconfig"

==> Analytics
install: 161,090 (30 days), 440,182 (90 days), 1,605,358 (365 days)
install-on-request: 96,767 (30 days), 305,863 (90 days), 1,222,200 (365 days)
build-error: 3,105 (30 days)
mindgood@minds-MacBook-Pro grpc_csharp_ext.dir % 
```

NDK have installed by Android Studio:
```
mindgood@minds-MacBook-Pro ~ % ndk-build   
Android NDK: Could not find application project directory !    
Android NDK: Please define the NDK_PROJECT_PATH variable to point to it.    
/Users/mindgood/Library/Android/sdk/ndk/25.1.8937393/build/core/build-local.mk:151: *** Android NDK: Aborting    .  Stop.
mindgood@minds-MacBook-Pro ~ % ndk-build -h
Usage: make [options] [target] ...
Options:
  -b, -m                      Ignored for compatibility.
  -B, --always-make           Unconditionally make all targets.
  -C DIRECTORY, --directory=DIRECTORY
                              Change to DIRECTORY before doing anything.
  -d                          Print lots of debugging information.
  --debug[=FLAGS]             Print various types of debugging information.
  -e, --environment-overrides
                              Environment variables override makefiles.
  -E STRING, --eval=STRING    Evaluate STRING as a makefile statement.
  -f FILE, --file=FILE, --makefile=FILE
                              Read FILE as a makefile.
  -h, --help                  Print this message and exit.
  -i, --ignore-errors         Ignore errors from recipes.
  -I DIRECTORY, --include-dir=DIRECTORY
                              Search DIRECTORY for included makefiles.
  -j [N], --jobs[=N]          Allow N jobs at once; infinite jobs with no arg.
  -k, --keep-going            Keep going when some targets can't be made.
  -l [N], --load-average[=N], --max-load[=N]
                              Don't start multiple jobs unless load is below N.
  -L, --check-symlink-times   Use the latest mtime between symlinks and target.
  -n, --just-print, --dry-run, --recon
                              Don't actually run any recipe; just print them.
  -o FILE, --old-file=FILE, --assume-old=FILE
                              Consider FILE to be very old and don't remake it.
  -O[TYPE], --output-sync[=TYPE]
                              Synchronize output of parallel jobs by TYPE.
  -p, --print-data-base       Print make's internal database.
  -q, --question              Run no recipe; exit status says if up to date.
  -r, --no-builtin-rules      Disable the built-in implicit rules.
  -R, --no-builtin-variables  Disable the built-in variable settings.
  -s, --silent, --quiet       Don't echo recipes.
  --no-silent                 Echo recipes (disable --silent mode).
  -S, --no-keep-going, --stop
                              Turns off -k.
  -t, --touch                 Touch targets instead of remaking them.
  --trace                     Print tracing information.
  -v, --version               Print the version number of make and exit.
  -w, --print-directory       Print the current directory.
  --no-print-directory        Turn off -w, even if it was turned on implicitly.
  -W FILE, --what-if=FILE, --new-file=FILE, --assume-new=FILE
                              Consider FILE to be infinitely new.
  --warn-undefined-variables  Warn when an undefined variable is referenced.

This program built for x86_64-apple-darwin19.3.0
Report bugs to <bug-make@gnu.org>
mindgood@minds-MacBook-Pro ~ % 
```

My ~/.zshrc is 
```
mindgood@minds-MacBook-Pro ~ %  cat ~/.zshrc
export ANDROID_HOME=/Users/mindgood/Library/Android/sdk
export PATH=$ANDROID_HOME/platform-tools:$PATH
export PATH=$ANDROID_HOME/tools:$PATH
export PATH=$ANDROID_HOME/tools/bin:$PATH 
export ANDROID_NDK=/Users/mindgood/Library/Android/sdk/ndk/25.1.8937393
export PATH=$PATH:$ANDROID_NDK

export PATH="/usr/local/opt/openssl@3/bin:$PATH"
export CPATH=/usr/local/opt/openssl@3/include:"${CPATH}"
export PKG_CONFIG_PATH=/usr/local/opt/openssl@3/lib/pkgconfig:"${PKG_CONFIG_PATH}"
export LD_LIBRARY_PATH=/usr/local/opt/openssl@3/lib:"${LD_LIBRARY_PATH}"

export LDFLAGS="-L/usr/local/opt/openssl@3/lib"
export CPPFLAGS="-I/usr/local/opt/openssl@3/include"

export CSC_OPTIONS="-I/usr/local/opt/openssl@3/include -L/usr/local/opt/openssl@3/lib"
mindgood@minds-MacBook-Pro ~ %
```
My gcc version is 
```
mindgood@minds-MacBook-Pro ~ % gcc --version
Apple clang version 14.0.0 (clang-1400.0.29.102)
Target: x86_64-apple-darwin21.6.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
mindgood@minds-MacBook-Pro ~ % 
```
then I use below command:
```
mindgood@minds-MacBook-Pro ~ %  cd ~
mindgood@minds-MacBook-Pro ~ % pwd
/Users/mindgood
mindgood@minds-MacBook-Pro ~ % cd Downloads 
mindgood@minds-MacBook-Pro Downloads % pwd
/Users/mindgood/Downloads
mindgood@minds-MacBook-Pro Downloads % git clone --recurse-submodules -b v1.30.0 https://github.com/grpc/grpc 
mindgood@minds-MacBook-Pro Downloads % cd grpc
mindgood@minds-MacBook-Pro Downloads % git checkout -b v1.22.x remotes/origin/v1.22.x 
mindgood@minds-MacBook-Pro Downloads % cmake ../grpc -DCMAKE_TOOLCHAIN_FILE=${ANDROID_NDK}/build/cmake/android.toolchain.cmake -DgRPC_BUILD_CODEGEN=OFF -DANDROID_ABI=armeabi-v7a -DANDROID_PLATFORM=android-26 -DCMAKE_BUILD_TYPE=Release
mindgood@minds-MacBook-Pro Downloads % cmake --build .
```
EXPLAIN when run below command is OK:
```
mindgood@minds-MacBook-Pro grpc % cmake ../grpc -DCMAKE_TOOLCHAIN_FILE=${ANDROID_NDK}/build/cmake/android.toolchain.cmake -DgRPC_BUILD_CODEGEN=OFF -DANDROID_ABI=armeabi-v7a -DANDROID_PLATFORM=android-26 -DCMAKE_BUILD_TYPE=Release
CMake Deprecation Warning at CMakeLists.txt:24 (cmake_minimum_required):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


-- The C compiler identification is Clang 14.0.6
-- The CXX compiler identification is Clang 14.0.6
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Users/mindgood/Library/Android/sdk/ndk/25.1.8937393/toolchains/llvm/prebuilt/darwin-x86_64/bin/clang - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Users/mindgood/Library/Android/sdk/ndk/25.1.8937393/toolchains/llvm/prebuilt/darwin-x86_64/bin/clang++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Deprecation Warning at third_party/zlib/CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Check size of off64_t
-- Check size of off64_t - done
-- Looking for fseeko
-- Looking for fseeko - found
-- Looking for unistd.h
-- Looking for unistd.h - found
CMake Warning at cmake/zlib.cmake:32 (message):
  gRPC_INSTALL will be forced to FALSE because gRPC_ZLIB_PROVIDER is "module"
Call Stack (most recent call first):
  CMakeLists.txt:141 (include)


-- Looking for res_servicename in resolv
-- Looking for res_servicename in resolv - not found
-- Looking for gethostbyname in nsl
-- Looking for gethostbyname in nsl - not found
-- Looking for gethostbyname in socket
-- Looking for gethostbyname in socket - not found
-- Looking for socket in socket
-- Looking for socket in socket - not found
-- Looking for clock_gettime in rt
-- Looking for clock_gettime in rt - not found
-- Looking for include file sys/socket.h
-- Looking for include file sys/socket.h - found
-- Looking for include file arpa/inet.h
-- Looking for include file arpa/inet.h - found
-- Looking for include file arpa/nameser_compat.h
-- Looking for include file arpa/nameser_compat.h - found
-- Looking for include file arpa/nameser.h
-- Looking for include file arpa/nameser.h - found
-- Looking for include file assert.h
-- Looking for include file assert.h - found
-- Looking for include file errno.h
-- Looking for include file errno.h - found
-- Looking for include file fcntl.h
-- Looking for include file fcntl.h - found
-- Looking for include file inttypes.h
-- Looking for include file inttypes.h - found
-- Looking for include file limits.h
-- Looking for include file limits.h - found
-- Looking for include file malloc.h
-- Looking for include file malloc.h - found
-- Looking for include file memory.h
-- Looking for include file memory.h - found
-- Looking for include file netdb.h
-- Looking for include file netdb.h - found
-- Looking for include file netinet/in.h
-- Looking for include file netinet/in.h - found
-- Looking for include file netinet/tcp.h
-- Looking for include file netinet/tcp.h - found
-- Looking for include file net/if.h
-- Looking for include file net/if.h - found
-- Looking for include file signal.h
-- Looking for include file signal.h - found
-- Looking for include file socket.h
-- Looking for include file socket.h - not found
-- Looking for include file stdbool.h
-- Looking for include file stdbool.h - found
-- Looking for include file stdlib.h
-- Looking for include file stdlib.h - found
-- Looking for include file strings.h
-- Looking for include file strings.h - found
-- Looking for include file string.h
-- Looking for include file string.h - found
-- Looking for include file stropts.h
-- Looking for include file stropts.h - not found
-- Looking for include file sys/ioctl.h
-- Looking for include file sys/ioctl.h - found
-- Looking for include file sys/param.h
-- Looking for include file sys/param.h - found
-- Looking for include file sys/select.h
-- Looking for include file sys/select.h - found
-- Looking for include file sys/stat.h
-- Looking for include file sys/stat.h - found
-- Looking for include file sys/time.h
-- Looking for include file sys/time.h - found
-- Looking for include file sys/uio.h
-- Looking for include file sys/uio.h - found
-- Looking for include file time.h
-- Looking for include file time.h - found
-- Looking for include file dlfcn.h
-- Looking for include file dlfcn.h - found
-- Looking for include file unistd.h
-- Looking for include file unistd.h - found
-- Looking for include files winsock2.h, windows.h
-- Looking for include files winsock2.h, windows.h - not found
-- Looking for 3 include files winsock2.h, ..., windows.h
-- Looking for 3 include files winsock2.h, ..., windows.h - not found
-- Looking for include files winsock.h, windows.h
-- Looking for include files winsock.h, windows.h - not found
-- Looking for include file windows.h
-- Looking for include file windows.h - not found
-- Performing Test HAVE_SOCKLEN_T
-- Performing Test HAVE_SOCKLEN_T - Success
-- Performing Test HAVE_TYPE_SOCKET
-- Performing Test HAVE_TYPE_SOCKET - Failed
-- Performing Test HAVE_BOOL_T
-- Performing Test HAVE_BOOL_T - Success
-- Performing Test HAVE_SSIZE_T
-- Performing Test HAVE_SSIZE_T - Success
-- Performing Test HAVE_LONGLONG
-- Performing Test HAVE_LONGLONG - Success
-- Performing Test HAVE_SIG_ATOMIC_T
-- Performing Test HAVE_SIG_ATOMIC_T - Success
-- Performing Test HAVE_STRUCT_ADDRINFO
-- Performing Test HAVE_STRUCT_ADDRINFO - Success
-- Performing Test HAVE_STRUCT_IN6_ADDR
-- Performing Test HAVE_STRUCT_IN6_ADDR - Success
-- Performing Test HAVE_STRUCT_SOCKADDR_IN6
-- Performing Test HAVE_STRUCT_SOCKADDR_IN6 - Success
-- Performing Test HAVE_STRUCT_SOCKADDR_STORAGE
-- Performing Test HAVE_STRUCT_SOCKADDR_STORAGE - Success
-- Performing Test HAVE_STRUCT_TIMEVAL
-- Performing Test HAVE_STRUCT_TIMEVAL - Success
-- Looking for AF_INET6
-- Looking for AF_INET6 - found
-- Looking for O_NONBLOCK
-- Looking for O_NONBLOCK - found
-- Looking for FIONBIO
-- Looking for FIONBIO - found
-- Looking for SIOCGIFADDR
-- Looking for SIOCGIFADDR - found
-- Looking for MSG_NOSIGNAL
-- Looking for MSG_NOSIGNAL - found
-- Looking for PF_INET6
-- Looking for PF_INET6 - found
-- Looking for SO_NONBLOCK
-- Looking for SO_NONBLOCK - not found
-- Looking for CLOCK_MONOTONIC
-- Looking for CLOCK_MONOTONIC - found
-- Performing Test HAVE_SOCKADDR_IN6_SIN6_SCOPE_ID
-- Performing Test HAVE_SOCKADDR_IN6_SIN6_SCOPE_ID - Success
-- Performing Test HAVE_LL
-- Performing Test HAVE_LL - Success
-- Looking for bitncmp
-- Looking for bitncmp - not found
-- Looking for closesocket
-- Looking for closesocket - not found
-- Looking for CloseSocket
-- Looking for CloseSocket - not found
-- Looking for connect
-- Looking for connect - found
-- Looking for fcntl
-- Looking for fcntl - found
-- Looking for freeaddrinfo
-- Looking for freeaddrinfo - found
-- Looking for getaddrinfo
-- Looking for getaddrinfo - found
-- Looking for getenv
-- Looking for getenv - found
-- Looking for gethostbyaddr
-- Looking for gethostbyaddr - found
-- Looking for gethostbyname
-- Looking for gethostbyname - found
-- Looking for gethostname
-- Looking for gethostname - found
-- Looking for getnameinfo
-- Looking for getnameinfo - found
-- Looking for getservbyport_r
-- Looking for getservbyport_r - not found
-- Looking for gettimeofday
-- Looking for gettimeofday - found
-- Looking for if_indextoname
-- Looking for if_indextoname - found
-- Looking for inet_net_pton
-- Looking for inet_net_pton - not found
-- Looking for inet_ntop
-- Looking for inet_ntop - found
-- Looking for inet_pton
-- Looking for inet_pton - found
-- Looking for ioctl
-- Looking for ioctl - not found
-- Looking for ioctlsocket
-- Looking for ioctlsocket - not found
-- Looking for IoctlSocket
-- Looking for IoctlSocket - not found
-- Looking for recv
-- Looking for recv - found
-- Looking for recvfrom
-- Looking for recvfrom - found
-- Looking for send
-- Looking for send - found
-- Looking for setsockopt
-- Looking for setsockopt - found
-- Looking for socket
-- Looking for socket - found
-- Looking for strcasecmp
-- Looking for strcasecmp - found
-- Looking for strcmpi
-- Looking for strcmpi - not found
-- Looking for strdup
-- Looking for strdup - found
-- Looking for stricmp
-- Looking for stricmp - not found
-- Looking for strncasecmp
-- Looking for strncasecmp - found
-- Looking for strncmpi
-- Looking for strncmpi - not found
-- Looking for strnicmp
-- Looking for strnicmp - not found
-- Looking for writev
-- Looking for writev - found
-- Looking for __system_property_get
-- Looking for __system_property_get - found
-- 
-- 3.12.2.0
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - not found
-- Check if compiler accepts -pthread
-- Check if compiler accepts -pthread - yes
-- Found Threads: TRUE  
-- Performing Test protobuf_HAVE_BUILTIN_ATOMICS
-- Performing Test protobuf_HAVE_BUILTIN_ATOMICS - Success
CMake Warning at cmake/ssl.cmake:34 (message):
  gRPC_SSL_PROVIDER is "module" but BORINGSSL_ROOT_DIR is wrong
Call Stack (most recent call first):
  CMakeLists.txt:144 (include)


-- Looking for C++ include fnmatch.h
-- Looking for C++ include fnmatch.h - found
-- Check size of uint32_t
-- Check size of uint32_t - done
-- Looking for strtoll
-- Looking for strtoll - found
-- Found Git: /usr/bin/git (found version "2.37.0 (Apple Git-136)") 
-- git Version: v1.5.0
-- Version: 1.5.0
-- Performing Test HAVE_CXX_FLAG_STD_CXX11
-- Performing Test HAVE_CXX_FLAG_STD_CXX11 - Success
-- Performing Test HAVE_CXX_FLAG_WALL
-- Performing Test HAVE_CXX_FLAG_WALL - Success
-- Performing Test HAVE_CXX_FLAG_WEXTRA
-- Performing Test HAVE_CXX_FLAG_WEXTRA - Success
-- Performing Test HAVE_CXX_FLAG_WSHADOW
-- Performing Test HAVE_CXX_FLAG_WSHADOW - Success
-- Performing Test HAVE_CXX_FLAG_WERROR
-- Performing Test HAVE_CXX_FLAG_WERROR - Success
-- Performing Test HAVE_CXX_FLAG_PEDANTIC
-- Performing Test HAVE_CXX_FLAG_PEDANTIC - Success
-- Performing Test HAVE_CXX_FLAG_PEDANTIC_ERRORS
-- Performing Test HAVE_CXX_FLAG_PEDANTIC_ERRORS - Success
-- Performing Test HAVE_CXX_FLAG_WSHORTEN_64_TO_32
-- Performing Test HAVE_CXX_FLAG_WSHORTEN_64_TO_32 - Success
-- Performing Test HAVE_CXX_FLAG_FSTRICT_ALIASING
-- Performing Test HAVE_CXX_FLAG_FSTRICT_ALIASING - Success
-- Performing Test HAVE_CXX_FLAG_WNO_DEPRECATED_DECLARATIONS
-- Performing Test HAVE_CXX_FLAG_WNO_DEPRECATED_DECLARATIONS - Success
-- Performing Test HAVE_CXX_FLAG_WNO_DEPRECATED
-- Performing Test HAVE_CXX_FLAG_WNO_DEPRECATED - Success
-- Performing Test HAVE_CXX_FLAG_WSTRICT_ALIASING
-- Performing Test HAVE_CXX_FLAG_WSTRICT_ALIASING - Success
-- Performing Test HAVE_CXX_FLAG_WD654
-- Performing Test HAVE_CXX_FLAG_WD654 - Failed
-- Performing Test HAVE_CXX_FLAG_WTHREAD_SAFETY
-- Performing Test HAVE_CXX_FLAG_WTHREAD_SAFETY - Success
-- Performing Test HAVE_THREAD_SAFETY_ATTRIBUTES
-- Performing Test HAVE_THREAD_SAFETY_ATTRIBUTES -- failed to compile
-- Performing Test HAVE_CXX_FLAG_COVERAGE
-- Performing Test HAVE_CXX_FLAG_COVERAGE - Success
-- Performing Test HAVE_STD_REGEX
CMake Warning at third_party/benchmark/cmake/CXXFeatureCheck.cmake:38 (message):
  If you see build failures due to cross compilation, try setting
  HAVE_STD_REGEX to 0
Call Stack (most recent call first):
  third_party/benchmark/CMakeLists.txt:247 (cxx_feature_check)


-- Performing Test HAVE_STD_REGEX -- success
-- Performing Test HAVE_GNU_POSIX_REGEX
-- Performing Test HAVE_GNU_POSIX_REGEX -- failed to compile
-- Performing Test HAVE_POSIX_REGEX
CMake Warning at third_party/benchmark/cmake/CXXFeatureCheck.cmake:38 (message):
  If you see build failures due to cross compilation, try setting
  HAVE_POSIX_REGEX to 0
Call Stack (most recent call first):
  third_party/benchmark/CMakeLists.txt:249 (cxx_feature_check)


-- Performing Test HAVE_POSIX_REGEX -- success
-- Performing Test HAVE_STEADY_CLOCK
CMake Warning at third_party/benchmark/cmake/CXXFeatureCheck.cmake:38 (message):
  If you see build failures due to cross compilation, try setting
  HAVE_STEADY_CLOCK to 0
Call Stack (most recent call first):
  third_party/benchmark/CMakeLists.txt:257 (cxx_feature_check)


-- Performing Test HAVE_STEADY_CLOCK -- success
-- Performing Test BENCHMARK_HAS_O3_FLAG
-- Performing Test BENCHMARK_HAS_O3_FLAG - Success
-- Performing Test BENCHMARK_HAS_CXX03_FLAG
-- Performing Test BENCHMARK_HAS_CXX03_FLAG - Success
-- Performing Test BENCHMARK_HAS_WNO_ODR
-- Performing Test BENCHMARK_HAS_WNO_ODR - Success
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/mindgood/Downloads/grpc
mindgood@minds-MacBook-Pro ~ %
```
But when run below command will get error:
```
mindgood@minds-MacBook-Pro grpc % cmake --build .
[  0%] Building C object CMakeFiles/address_sorting.dir/third_party/address_sorting/address_sorting.c.o
[  0%] Building C object CMakeFiles/address_sorting.dir/third_party/address_sorting/address_sorting_posix.c.o
[  0%] Building C object CMakeFiles/address_sorting.dir/third_party/address_sorting/address_sorting_windows.c.o
[  0%] Linking C static library libaddress_sorting.a
[  0%] Built target address_sorting
[  0%] Building CXX object CMakeFiles/gpr.dir/src/core/lib/gpr/alloc.cc.o
[  0%] Building CXX object CMakeFiles/gpr.dir/src/core/lib/gpr/atm.cc.o
[  0%] Building CXX object CMakeFiles/gpr.dir/src/core/lib/gpr/cpu_iphone.cc.o
[  0%] Building CXX object CMakeFiles/gpr.dir/src/core/lib/gpr/cpu_linux.cc.o
[  0%] Building CXX object CMakeFiles/gpr.dir/src/core/lib/gpr/cpu_posix.cc.o
[  0%] Building CXX object CMakeFiles/gpr.dir/src/core/lib/gpr/cpu_windows.cc.o
[  0%] Building CXX object CMakeFiles/gpr.dir/src/core/lib/gpr/env_linux.cc.o
[  0%] Building CXX object CMakeFiles/gpr.dir/src/core/lib/gpr/env_posix.cc.o
[  0%] Building CXX object CMakeFiles/gpr.dir/src/core/lib/gpr/env_windows.cc.o
[  0%] Building CXX object CMakeFiles/gpr.dir/src/core/lib/gpr/host_port.cc.o
[  0%] Building CXX object CMakeFiles/gpr.dir/src/core/lib/gpr/log.cc.o
[  1%] Building CXX object CMakeFiles/gpr.dir/src/core/lib/gpr/log_android.cc.o
[  1%] Building CXX object CMakeFiles/gpr.dir/src/core/lib/gpr/log_linux.cc.o
[  1%] Building CXX object CMakeFiles/gpr.dir/src/core/lib/gpr/log_posix.cc.o
[  1%] Building CXX object CMakeFiles/gpr.dir/src/core/lib/gpr/log_windows.cc.o
[  1%] Building CXX object CMakeFiles/gpr.dir/src/core/lib/gpr/mpscq.cc.o
[  1%] Building CXX object CMakeFiles/gpr.dir/src/core/lib/gpr/murmur_hash.cc.o
[  1%] Building CXX object CMakeFiles/gpr.dir/src/core/lib/gpr/string.cc.o
[  1%] Building CXX object CMakeFiles/gpr.dir/src/core/lib/gpr/string_posix.cc.o
[  1%] Building CXX object CMakeFiles/gpr.dir/src/core/lib/gpr/string_util_windows.cc.o
[  1%] Building CXX object CMakeFiles/gpr.dir/src/core/lib/gpr/string_windows.cc.o
[  1%] Building CXX object CMakeFiles/gpr.dir/src/core/lib/gpr/sync.cc.o
[  1%] Building CXX object CMakeFiles/gpr.dir/src/core/lib/gpr/sync_posix.cc.o
[  1%] Building CXX object CMakeFiles/gpr.dir/src/core/lib/gpr/sync_windows.cc.o
[  1%] Building CXX object CMakeFiles/gpr.dir/src/core/lib/gpr/time.cc.o
[  1%] Building CXX object CMakeFiles/gpr.dir/src/core/lib/gpr/time_posix.cc.o
[  1%] Building CXX object CMakeFiles/gpr.dir/src/core/lib/gpr/time_precise.cc.o
[  1%] Building CXX object CMakeFiles/gpr.dir/src/core/lib/gpr/time_windows.cc.o
[  2%] Building CXX object CMakeFiles/gpr.dir/src/core/lib/gpr/tls_pthread.cc.o
[  2%] Building CXX object CMakeFiles/gpr.dir/src/core/lib/gpr/tmpfile_msys.cc.o
[  2%] Building CXX object CMakeFiles/gpr.dir/src/core/lib/gpr/tmpfile_posix.cc.o
[  2%] Building CXX object CMakeFiles/gpr.dir/src/core/lib/gpr/tmpfile_windows.cc.o
[  2%] Building CXX object CMakeFiles/gpr.dir/src/core/lib/gpr/wrap_memcpy.cc.o
[  2%] Building CXX object CMakeFiles/gpr.dir/src/core/lib/gprpp/arena.cc.o
[  2%] Building CXX object CMakeFiles/gpr.dir/src/core/lib/gprpp/fork.cc.o
[  2%] Building CXX object CMakeFiles/gpr.dir/src/core/lib/gprpp/global_config_env.cc.o
[  2%] Building CXX object CMakeFiles/gpr.dir/src/core/lib/gprpp/thd_posix.cc.o
[  2%] Building CXX object CMakeFiles/gpr.dir/src/core/lib/gprpp/thd_windows.cc.o
[  2%] Building CXX object CMakeFiles/gpr.dir/src/core/lib/profiling/basic_timers.cc.o
[  2%] Building CXX object CMakeFiles/gpr.dir/src/core/lib/profiling/stap_timers.cc.o
[  2%] Linking CXX static library libgpr.a
[  2%] Built target gpr
[  2%] Building C object third_party/cares/cares/CMakeFiles/c-ares.dir/ares__close_sockets.c.o
[  2%] Building C object third_party/cares/cares/CMakeFiles/c-ares.dir/ares__get_hostent.c.o
[  2%] Building C object third_party/cares/cares/CMakeFiles/c-ares.dir/ares__read_line.c.o
[  2%] Building C object third_party/cares/cares/CMakeFiles/c-ares.dir/ares__timeval.c.o
[  2%] Building C object third_party/cares/cares/CMakeFiles/c-ares.dir/ares_android.c.o
[  2%] Building C object third_party/cares/cares/CMakeFiles/c-ares.dir/ares_cancel.c.o
[  2%] Building C object third_party/cares/cares/CMakeFiles/c-ares.dir/ares_data.c.o
[  3%] Building C object third_party/cares/cares/CMakeFiles/c-ares.dir/ares_destroy.c.o
[  3%] Building C object third_party/cares/cares/CMakeFiles/c-ares.dir/ares_expand_name.c.o
[  3%] Building C object third_party/cares/cares/CMakeFiles/c-ares.dir/ares_expand_string.c.o
[  3%] Building C object third_party/cares/cares/CMakeFiles/c-ares.dir/ares_fds.c.o
[  3%] Building C object third_party/cares/cares/CMakeFiles/c-ares.dir/ares_free_hostent.c.o
[  3%] Building C object third_party/cares/cares/CMakeFiles/c-ares.dir/ares_free_string.c.o
[  3%] Building C object third_party/cares/cares/CMakeFiles/c-ares.dir/ares_getenv.c.o
[  3%] Building C object third_party/cares/cares/CMakeFiles/c-ares.dir/ares_gethostbyaddr.c.o
[  3%] Building C object third_party/cares/cares/CMakeFiles/c-ares.dir/ares_gethostbyname.c.o
[  3%] Building C object third_party/cares/cares/CMakeFiles/c-ares.dir/ares_getnameinfo.c.o
[  3%] Building C object third_party/cares/cares/CMakeFiles/c-ares.dir/ares_getsock.c.o
[  3%] Building C object third_party/cares/cares/CMakeFiles/c-ares.dir/ares_init.c.o
[  3%] Building C object third_party/cares/cares/CMakeFiles/c-ares.dir/ares_library_init.c.o
[  3%] Building C object third_party/cares/cares/CMakeFiles/c-ares.dir/ares_llist.c.o
[  3%] Building C object third_party/cares/cares/CMakeFiles/c-ares.dir/ares_mkquery.c.o
[  3%] Building C object third_party/cares/cares/CMakeFiles/c-ares.dir/ares_create_query.c.o
[  3%] Building C object third_party/cares/cares/CMakeFiles/c-ares.dir/ares_nowarn.c.o
[  4%] Building C object third_party/cares/cares/CMakeFiles/c-ares.dir/ares_options.c.o
[  4%] Building C object third_party/cares/cares/CMakeFiles/c-ares.dir/ares_parse_a_reply.c.o
[  4%] Building C object third_party/cares/cares/CMakeFiles/c-ares.dir/ares_parse_aaaa_reply.c.o
[  4%] Building C object third_party/cares/cares/CMakeFiles/c-ares.dir/ares_parse_mx_reply.c.o
[  4%] Building C object third_party/cares/cares/CMakeFiles/c-ares.dir/ares_parse_naptr_reply.c.o
[  4%] Building C object third_party/cares/cares/CMakeFiles/c-ares.dir/ares_parse_ns_reply.c.o
[  4%] Building C object third_party/cares/cares/CMakeFiles/c-ares.dir/ares_parse_ptr_reply.c.o
[  4%] Building C object third_party/cares/cares/CMakeFiles/c-ares.dir/ares_parse_soa_reply.c.o
[  4%] Building C object third_party/cares/cares/CMakeFiles/c-ares.dir/ares_parse_srv_reply.c.o
[  4%] Building C object third_party/cares/cares/CMakeFiles/c-ares.dir/ares_parse_txt_reply.c.o
[  4%] Building C object third_party/cares/cares/CMakeFiles/c-ares.dir/ares_platform.c.o
[  4%] Building C object third_party/cares/cares/CMakeFiles/c-ares.dir/ares_process.c.o
[  4%] Building C object third_party/cares/cares/CMakeFiles/c-ares.dir/ares_query.c.o
[  4%] Building C object third_party/cares/cares/CMakeFiles/c-ares.dir/ares_search.c.o
[  4%] Building C object third_party/cares/cares/CMakeFiles/c-ares.dir/ares_send.c.o
[  4%] Building C object third_party/cares/cares/CMakeFiles/c-ares.dir/ares_strcasecmp.c.o
[  4%] Building C object third_party/cares/cares/CMakeFiles/c-ares.dir/ares_strdup.c.o
[  5%] Building C object third_party/cares/cares/CMakeFiles/c-ares.dir/ares_strerror.c.o
......
......
......
......
......
......
[ 89%] Building CXX object third_party/protobuf/CMakeFiles/libprotoc.dir/src/google/protobuf/compiler/objectivec/objectivec_extension.cc.o
[ 89%] Building CXX object third_party/protobuf/CMakeFiles/libprotoc.dir/src/google/protobuf/compiler/objectivec/objectivec_field.cc.o
[ 89%] Building CXX object third_party/protobuf/CMakeFiles/libprotoc.dir/src/google/protobuf/compiler/objectivec/objectivec_file.cc.o
[ 89%] Building CXX object third_party/protobuf/CMakeFiles/libprotoc.dir/src/google/protobuf/compiler/objectivec/objectivec_generator.cc.o
[ 89%] Building CXX object third_party/protobuf/CMakeFiles/libprotoc.dir/src/google/protobuf/compiler/objectivec/objectivec_helpers.cc.o
[ 89%] Building CXX object third_party/protobuf/CMakeFiles/libprotoc.dir/src/google/protobuf/compiler/objectivec/objectivec_map_field.cc.o
[ 89%] Building CXX object third_party/protobuf/CMakeFiles/libprotoc.dir/src/google/protobuf/compiler/objectivec/objectivec_message.cc.o
[ 89%] Building CXX object third_party/protobuf/CMakeFiles/libprotoc.dir/src/google/protobuf/compiler/objectivec/objectivec_message_field.cc.o
[ 89%] Building CXX object third_party/protobuf/CMakeFiles/libprotoc.dir/src/google/protobuf/compiler/objectivec/objectivec_oneof.cc.o
[ 89%] Building CXX object third_party/protobuf/CMakeFiles/libprotoc.dir/src/google/protobuf/compiler/objectivec/objectivec_primitive_field.cc.o
[ 89%] Building CXX object third_party/protobuf/CMakeFiles/libprotoc.dir/src/google/protobuf/compiler/php/php_generator.cc.o
[ 89%] Building CXX object third_party/protobuf/CMakeFiles/libprotoc.dir/src/google/protobuf/compiler/plugin.cc.o
[ 89%] Building CXX object third_party/protobuf/CMakeFiles/libprotoc.dir/src/google/protobuf/compiler/plugin.pb.cc.o
[ 89%] Building CXX object third_party/protobuf/CMakeFiles/libprotoc.dir/src/google/protobuf/compiler/python/python_generator.cc.o
[ 89%] Building CXX object third_party/protobuf/CMakeFiles/libprotoc.dir/src/google/protobuf/compiler/ruby/ruby_generator.cc.o
[ 90%] Building CXX object third_party/protobuf/CMakeFiles/libprotoc.dir/src/google/protobuf/compiler/subprocess.cc.o
[ 90%] Building CXX object third_party/protobuf/CMakeFiles/libprotoc.dir/src/google/protobuf/compiler/zip_writer.cc.o
[ 90%] Linking CXX static library libprotoc.a
[ 90%] Built target libprotoc
[ 90%] Building CXX object CMakeFiles/grpc_plugin_support.dir/src/compiler/cpp_generator.cc.o
[ 90%] Building CXX object CMakeFiles/grpc_plugin_support.dir/src/compiler/csharp_generator.cc.o
[ 90%] Building CXX object CMakeFiles/grpc_plugin_support.dir/src/compiler/node_generator.cc.o
[ 90%] Building CXX object CMakeFiles/grpc_plugin_support.dir/src/compiler/objective_c_generator.cc.o
[ 90%] Building CXX object CMakeFiles/grpc_plugin_support.dir/src/compiler/php_generator.cc.o
[ 90%] Building CXX object CMakeFiles/grpc_plugin_support.dir/src/compiler/python_generator.cc.o
[ 91%] Building CXX object CMakeFiles/grpc_plugin_support.dir/src/compiler/ruby_generator.cc.o
[ 91%] Linking CXX static library libgrpc_plugin_support.a
[ 91%] Built target grpc_plugin_support
[ 91%] Building C object CMakeFiles/grpc_csharp_ext.dir/src/csharp/ext/grpc_csharp_ext.c.o
[ 91%] Linking CXX shared library libgrpc_csharp_ext.so
ld: error: undefined symbol: X509_STORE_new
>>> referenced by ssl_transport_security.cc:787 (src/core/tsi/ssl_transport_security.cc:787)
>>>               ssl_transport_security.cc.o:(tsi_ssl_root_certs_store_create(char const*)) in archive libgrpc.a

ld: error: undefined symbol: X509_STORE_free
>>> referenced by ssl_transport_security.cc:797 (src/core/tsi/ssl_transport_security.cc:797)
>>>               ssl_transport_security.cc.o:(tsi_ssl_root_certs_store_create(char const*)) in archive libgrpc.a
>>> referenced by ssl_transport_security.cc:806 (src/core/tsi/ssl_transport_security.cc:806)
>>>               ssl_transport_security.cc.o:(tsi_ssl_root_certs_store_destroy(tsi_ssl_root_certs_store*)) in archive libgrpc.a

ld: error: undefined symbol: BIO_new_mem_buf
>>> referenced by ssl_transport_security.cc:593 (src/core/tsi/ssl_transport_security.cc:593)
>>>               ssl_transport_security.cc.o:(x509_store_load_certs(x509_store_st*, char const*, unsigned int, stack_st_X509_NAME**)) in archive libgrpc.a
>>> referenced by ssl_transport_security.cc:519 (src/core/tsi/ssl_transport_security.cc:519)
>>>               ssl_transport_security.cc.o:(populate_ssl_context(ssl_ctx_st*, tsi_ssl_pem_key_cert_pair const*, char const*)) in archive libgrpc.a
>>> referenced by ssl_transport_security.cc:563 (src/core/tsi/ssl_transport_security.cc:563)
>>>               ssl_transport_security.cc.o:(populate_ssl_context(ssl_ctx_st*, tsi_ssl_pem_key_cert_pair const*, char const*)) in archive libgrpc.a
>>> referenced 1 more times

ld: error: undefined symbol: OPENSSL_sk_new_null
>>> referenced by ssl_transport_security.cc:597 (src/core/tsi/ssl_transport_security.cc:597)
>>>               ssl_transport_security.cc.o:(x509_store_load_certs(x509_store_st*, char const*, unsigned int, stack_st_X509_NAME**)) in archive libgrpc.a

ld: error: undefined symbol: PEM_read_bio_X509_AUX
>>> referenced by ssl_transport_security.cc:602 (src/core/tsi/ssl_transport_security.cc:602)
>>>               ssl_transport_security.cc.o:(x509_store_load_certs(x509_store_st*, char const*, unsigned int, stack_st_X509_NAME**)) in archive libgrpc.a
>>> referenced by ssl_transport_security.cc:602 (src/core/tsi/ssl_transport_security.cc:602)
>>>               ssl_transport_security.cc.o:(x509_store_load_certs(x509_store_st*, char const*, unsigned int, stack_st_X509_NAME**)) in archive libgrpc.a
>>> referenced by ssl_transport_security.cc:602 (src/core/tsi/ssl_transport_security.cc:602)
>>>               ssl_transport_security.cc.o:(x509_store_load_certs(x509_store_st*, char const*, unsigned int, stack_st_X509_NAME**)) in archive libgrpc.a
>>> referenced 2 more times

ld: error: undefined symbol: X509_free
>>> referenced by ssl_transport_security.cc:632 (src/core/tsi/ssl_transport_security.cc:632)
>>>               ssl_transport_security.cc.o:(x509_store_load_certs(x509_store_st*, char const*, unsigned int, stack_st_X509_NAME**)) in archive libgrpc.a
>>> referenced by ssl_transport_security.cc:632 (src/core/tsi/ssl_transport_security.cc:632)
>>>               ssl_transport_security.cc.o:(x509_store_load_certs(x509_store_st*, char const*, unsigned int, stack_st_X509_NAME**)) in archive libgrpc.a
>>> referenced by ssl_transport_security.cc:641 (src/core/tsi/ssl_transport_security.cc:641)
>>>               ssl_transport_security.cc.o:(x509_store_load_certs(x509_store_st*, char const*, unsigned int, stack_st_X509_NAME**)) in archive libgrpc.a
>>> referenced 4 more times

ld: error: undefined symbol: X509_get_subject_name
>>> referenced by ssl_transport_security.cc:608 (src/core/tsi/ssl_transport_security.cc:608)
>>>               ssl_transport_security.cc.o:(x509_store_load_certs(x509_store_st*, char const*, unsigned int, stack_st_X509_NAME**)) in archive libgrpc.a
>>> referenced by ssl_transport_security.cc:266 (src/core/tsi/ssl_transport_security.cc:266)
>>>               ssl_transport_security.cc.o:(peer_from_x509(x509_st*, int, tsi_peer*)) in archive libgrpc.a

ld: error: undefined symbol: X509_NAME_dup
>>> referenced by ssl_transport_security.cc:614 (src/core/tsi/ssl_transport_security.cc:614)
>>>               ssl_transport_security.cc.o:(x509_store_load_certs(x509_store_st*, char const*, unsigned int, stack_st_X509_NAME**)) in archive libgrpc.a

ld: error: undefined symbol: OPENSSL_sk_push
>>> referenced by ssl_transport_security.cc:619 (src/core/tsi/ssl_transport_security.cc:619)
>>>               ssl_transport_security.cc.o:(x509_store_load_certs(x509_store_st*, char const*, unsigned int, stack_st_X509_NAME**)) in archive libgrpc.a

ld: error: undefined symbol: ERR_clear_error
>>> referenced by ssl_transport_security.cc:622 (src/core/tsi/ssl_transport_security.cc:622)
>>>               ssl_transport_security.cc.o:(x509_store_load_certs(x509_store_st*, char const*, unsigned int, stack_st_X509_NAME**)) in archive libgrpc.a
>>> referenced by ssl_transport_security.cc:622 (src/core/tsi/ssl_transport_security.cc:622)
>>>               ssl_transport_security.cc.o:(x509_store_load_certs(x509_store_st*, char const*, unsigned int, stack_st_X509_NAME**)) in archive libgrpc.a
>>> referenced by ssl_transport_security.cc:604 (src/core/tsi/ssl_transport_security.cc:604)
>>>               ssl_transport_security.cc.o:(x509_store_load_certs(x509_store_st*, char const*, unsigned int, stack_st_X509_NAME**)) in archive libgrpc.a
>>> referenced 2 more times

ld: error: undefined symbol: X509_STORE_add_cert
>>> referenced by ssl_transport_security.cc:623 (src/core/tsi/ssl_transport_security.cc:623)
>>>               ssl_transport_security.cc.o:(x509_store_load_certs(x509_store_st*, char const*, unsigned int, stack_st_X509_NAME**)) in archive libgrpc.a
>>> referenced by ssl_transport_security.cc:623 (src/core/tsi/ssl_transport_security.cc:623)
>>>               ssl_transport_security.cc.o:(x509_store_load_certs(x509_store_st*, char const*, unsigned int, stack_st_X509_NAME**)) in archive libgrpc.a

ld: error: undefined symbol: ERR_get_error
>>> referenced by ssl_transport_security.cc:624 (src/core/tsi/ssl_transport_security.cc:624)
>>>               ssl_transport_security.cc.o:(x509_store_load_certs(x509_store_st*, char const*, unsigned int, stack_st_X509_NAME**)) in archive libgrpc.a
>>> referenced by ssl_transport_security.cc:624 (src/core/tsi/ssl_transport_security.cc:624)
>>>               ssl_transport_security.cc.o:(x509_store_load_certs(x509_store_st*, char const*, unsigned int, stack_st_X509_NAME**)) in archive libgrpc.a
>>> referenced by ssl_transport_security.cc:1237 (src/core/tsi/ssl_transport_security.cc:1237)
>>>               ssl_transport_security.cc.o:(ssl_handshaker_next(tsi_handshaker*, unsigned char const*, unsigned int, unsigned char const**, unsigned int*, tsi_handshaker_result**, void (*)(tsi_result, void*, unsigned char const*, unsigned int, tsi_handshaker_result*), void*)) in archive libgrpc.a
>>> referenced 2 more times

ld: error: undefined symbol: OPENSSL_sk_pop_free
>>> referenced by ssl_transport_security.cc:643 (src/core/tsi/ssl_transport_security.cc:643)
>>>               ssl_transport_security.cc.o:(x509_store_load_certs(x509_store_st*, char const*, unsigned int, stack_st_X509_NAME**)) in archive libgrpc.a
>>> referenced by ssl_transport_security.cc:440 (src/core/tsi/ssl_transport_security.cc:440)
>>>               ssl_transport_security.cc.o:(peer_from_x509(x509_st*, int, tsi_peer*)) in archive libgrpc.a

ld: error: undefined symbol: BIO_free
>>> referenced by ssl_transport_security.cc:648 (src/core/tsi/ssl_transport_security.cc:648)
>>>               ssl_transport_security.cc.o:(x509_store_load_certs(x509_store_st*, char const*, unsigned int, stack_st_X509_NAME**)) in archive libgrpc.a
>>> referenced by ssl_transport_security.cc:0 (src/core/tsi/ssl_transport_security.cc:0)
>>>               ssl_transport_security.cc.o:(create_tsi_ssl_handshaker(ssl_ctx_st*, int, char const*, tsi_ssl_handshaker_factory*, tsi_handshaker**)) in archive libgrpc.a
>>> referenced by ssl_transport_security.cc:552 (src/core/tsi/ssl_transport_security.cc:552)
>>>               ssl_transport_security.cc.o:(populate_ssl_context(ssl_ctx_st*, tsi_ssl_pem_key_cert_pair const*, char const*)) in archive libgrpc.a
>>> referenced 10 more times

ld: error: undefined symbol: X509_NAME_free
>>> referenced by ssl_transport_security.cc:0 (src/core/tsi/ssl_transport_security.cc:0)
>>>               ssl_transport_security.cc.o:(x509_store_load_certs(x509_store_st*, char const*, unsigned int, stack_st_X509_NAME**)) in archive libgrpc.a

ld: error: undefined symbol: SSL_new
>>> referenced by ssl_transport_security.cc:1340 (src/core/tsi/ssl_transport_security.cc:1340)
>>>               ssl_transport_security.cc.o:(create_tsi_ssl_handshaker(ssl_ctx_st*, int, char const*, tsi_ssl_handshaker_factory*, tsi_handshaker**)) in archive libgrpc.a

ld: error: undefined symbol: SSL_set_info_callback
>>> referenced by ssl_transport_security.cc:1352 (src/core/tsi/ssl_transport_security.cc:1352)
>>>               ssl_transport_security.cc.o:(create_tsi_ssl_handshaker(ssl_ctx_st*, int, char const*, tsi_ssl_handshaker_factory*, tsi_handshaker**)) in archive libgrpc.a

ld: error: undefined symbol: BIO_new_bio_pair
>>> referenced by ssl_transport_security.cc:1354 (src/core/tsi/ssl_transport_security.cc:1354)
>>>               ssl_transport_security.cc.o:(create_tsi_ssl_handshaker(ssl_ctx_st*, int, char const*, tsi_ssl_handshaker_factory*, tsi_handshaker**)) in archive libgrpc.a

ld: error: undefined symbol: SSL_set_bio
>>> referenced by ssl_transport_security.cc:1359 (src/core/tsi/ssl_transport_security.cc:1359)
>>>               ssl_transport_security.cc.o:(create_tsi_ssl_handshaker(ssl_ctx_st*, int, char const*, tsi_ssl_handshaker_factory*, tsi_handshaker**)) in archive libgrpc.a

ld: error: undefined symbol: SSL_set_connect_state
>>> referenced by ssl_transport_security.cc:1362 (src/core/tsi/ssl_transport_security.cc:1362)
>>>               ssl_transport_security.cc.o:(create_tsi_ssl_handshaker(ssl_ctx_st*, int, char const*, tsi_ssl_handshaker_factory*, tsi_handshaker**)) in archive libgrpc.a

ld: error: too many errors emitted, stopping now (use -error-limit=0 to see all errors)
clang++: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [libgrpc_csharp_ext.so] Error 1
make[1]: *** [CMakeFiles/grpc_csharp_ext.dir/all] Error 2
make: *** [all] Error 2
mindgood@minds-MacBook-Pro grpc % 
```

We need to solve X509 and openssl error:
```
mindgood@minds-MacBook-Pro grpc % cd CMakeFiles/grpc_csharp_ext.dir 
mindgood@minds-MacBook-Pro grpc_csharp_ext.dir % pwd
/Users/mindgood/Downloads/grpc/CMakeFiles/grpc_csharp_ext.dir
mindgood@minds-MacBook-Pro grpc_csharp_ext.dir % ls -la
total 80
drwxr-xr-x  12 mindgood  staff   384 Oct 21 09:51 .
drwxr-xr-x  48 mindgood  staff  1536 Oct 21 09:53 ..
-rw-r--r--   1 mindgood  staff  1054 Oct 17 20:55 DependInfo.cmake
-rw-r--r--   1 mindgood  staff  6058 Oct 20 11:44 build.make
-rw-r--r--   1 mindgood  staff   377 Oct 21 09:51 cmake_clean.cmake
-rw-r--r--   1 mindgood  staff   118 Oct 21 09:51 compiler_depend.make
-rw-r--r--   1 mindgood  staff   122 Oct 17 20:55 compiler_depend.ts
-rw-r--r--   1 mindgood  staff    99 Oct 21 09:51 depend.make
-rw-r--r--   1 mindgood  staff  1016 Oct 21 09:51 flags.make
-rw-r--r--   1 mindgood  staff   896 Oct 21 09:51 link.txt
-rw-r--r--   1 mindgood  staff    41 Oct 21 09:51 progress.make
drwxr-xr-x   3 mindgood  staff    96 Oct 17 20:55 src
mindgood@minds-MacBook-Pro grpc_csharp_ext.dir % cat link.txt 
/Users/mindgood/Library/Android/sdk/ndk/25.1.8937393/toolchains/llvm/prebuilt/darwin-x86_64/bin/clang++ --target=armv7-none-linux-androideabi26 --sysroot=/Users/mindgood/Library/Android/sdk/ndk/25.1.8937393/toolchains/llvm/prebuilt/darwin-x86_64/sysroot -fPIC -g -DANDROID -fdata-sections -ffunction-sections -funwind-tables -fstack-protector-strong -no-canonical-prefixes -D_FORTIFY_SOURCE=2 -march=armv7-a -mthumb -Wformat -Werror=format-security   -std=c++11 -O3 -DNDEBUG  -static-libstdc++ -Wl,--build-id=sha1 -Wl,--no-rosegment -Wl,--fatal-warnings -Wl,--gc-sections -Wl,--no-undefined -Qunused-arguments  -shared -Wl,-soname,libgrpc_csharp_ext.so -o libgrpc_csharp_ext.so CMakeFiles/grpc_csharp_ext.dir/src/csharp/ext/grpc_csharp_ext.c.o  -ldl -lm libgrpc.a libgpr.a -landroid -llog third_party/zlib/libz.a third_party/cares/cares/lib/libcares.a libaddress_sorting.a -ldl -lm -latomic -lm 
mindgood@minds-MacBook-Pro grpc_csharp_ext.dir % 
```

Now need change link.txt to be like below:
```
mindgood@minds-MacBook-Pro grpc_csharp_ext.dir % cat link.txt 
/Users/mindgood/Library/Android/sdk/ndk/25.1.8937393/toolchains/llvm/prebuilt/darwin-x86_64/bin/clang++ --target=armv7-none-linux-androideabi26 --sysroot=/Users/mindgood/Library/Android/sdk/ndk/25.1.8937393/toolchains/llvm/prebuilt/darwin-x86_64/sysroot -fPIC -g -DANDROID -fdata-sections -ffunction-sections -funwind-tables -fstack-protector-strong -no-canonical-prefixes -D_FORTIFY_SOURCE=2 -march=armv7-a -mthumb -Wformat -Werror=format-security   -std=c++11 -O3 -DNDEBUG  -static-libstdc++ -Wl,--build-id=sha1 -Wl,--no-rosegment -Wl,--fatal-warnings -Wl,--gc-sections -Wl,--no-undefined -Qunused-arguments  -shared -Wl,-soname,libgrpc_csharp_ext.so -o libgrpc_csharp_ext.so CMakeFiles/grpc_csharp_ext.dir/src/csharp/ext/grpc_csharp_ext.c.o  -ldl -lm libgrpc.a libgpr.a -L  -landroid -L -llog third_party/zlib/libz.a third_party/cares/cares/lib/libcares.a libaddress_sorting.a -ldl -lm -latomic -lm -static  -L -lssl -L -lcrypto 
mindgood@minds-MacBook-Pro grpc_csharp_ext.dir % 
```
General speaking:
add below argument:
```
-static  -L -lssl -L -lcrypto 
```
in end of link.txt,

and add -L in front of -landroid, add -L in front of -llog.

Then run again below command:
```
mindgood@minds-MacBook-Pro grpc % cmake --build .
Consolidate compiler generated dependencies of target address_sorting
[  0%] Built target address_sorting
Consolidate compiler generated dependencies of target gpr
[  2%] Built target gpr
Consolidate compiler generated dependencies of target c-ares
[  5%] Built target c-ares
Consolidate compiler generated dependencies of target zlibstatic
[  6%] Built target zlibstatic
Consolidate compiler generated dependencies of target grpc
[ 26%] Built target grpc
Consolidate compiler generated dependencies of target grpc_cronet
[ 44%] Built target grpc_cronet
Consolidate compiler generated dependencies of target grpc_unsecure
[ 59%] Built target grpc_unsecure
Consolidate compiler generated dependencies of target libprotobuf
[ 64%] Built target libprotobuf
Consolidate compiler generated dependencies of target grpc++
[ 67%] Built target grpc++
Consolidate compiler generated dependencies of target grpc++_cronet
[ 83%] Built target grpc++_cronet
Consolidate compiler generated dependencies of target grpc++_unsecure
[ 85%] Built target grpc++_unsecure
Consolidate compiler generated dependencies of target libprotoc
[ 90%] Built target libprotoc
Consolidate compiler generated dependencies of target grpc_plugin_support
[ 91%] Built target grpc_plugin_support
Consolidate compiler generated dependencies of target grpc_csharp_ext
[ 91%] Linking CXX shared library libgrpc_csharp_ext.so
ld: error: /Users/mindgood/Library/Android/sdk/ndk/25.1.8937393/toolchains/llvm/prebuilt/darwin-x86_64/lib64/clang/14.0.6/lib/linux/libclang_rt.builtins-arm-android.a(aeabi_memset.S.o): symbol __aeabi_memclr@@LIBC_N has undefined version LIBC_N
ld: error: /Users/mindgood/Library/Android/sdk/ndk/25.1.8937393/toolchains/llvm/prebuilt/darwin-x86_64/lib64/clang/14.0.6/lib/linux/libclang_rt.builtins-arm-android.a(aeabi_memset.S.o): symbol __aeabi_memclr8@@LIBC_N has undefined version LIBC_N
ld: error: /Users/mindgood/Library/Android/sdk/ndk/25.1.8937393/toolchains/llvm/prebuilt/darwin-x86_64/lib64/clang/14.0.6/lib/linux/libclang_rt.builtins-arm-android.a(aeabi_memcpy.S.o): symbol __aeabi_memcpy@@LIBC_N has undefined version LIBC_N
ld: error: /Users/mindgood/Library/Android/sdk/ndk/25.1.8937393/toolchains/llvm/prebuilt/darwin-x86_64/lib64/clang/14.0.6/lib/linux/libclang_rt.builtins-arm-android.a(aeabi_memmove.S.o): symbol __aeabi_memmove@@LIBC_N has undefined version LIBC_N
ld: error: /Users/mindgood/Library/Android/sdk/ndk/25.1.8937393/toolchains/llvm/prebuilt/darwin-x86_64/lib64/clang/14.0.6/lib/linux/libclang_rt.builtins-arm-android.a(aeabi_memcpy.S.o): symbol __aeabi_memcpy8@@LIBC_N has undefined version LIBC_N
ld: error: /Users/mindgood/Library/Android/sdk/ndk/25.1.8937393/toolchains/llvm/prebuilt/darwin-x86_64/lib64/clang/14.0.6/lib/linux/libclang_rt.builtins-arm-android.a(aeabi_memset.S.o): symbol __aeabi_memset@@LIBC_N has undefined version LIBC_N
ld: error: /Users/mindgood/Library/Android/sdk/ndk/25.1.8937393/toolchains/llvm/prebuilt/darwin-x86_64/lib64/clang/14.0.6/lib/linux/libclang_rt.builtins-arm-android.a(aeabi_memcpy.S.o): symbol __aeabi_memcpy4@@LIBC_N has undefined version LIBC_N
ld: error: /Users/mindgood/Library/Android/sdk/ndk/25.1.8937393/toolchains/llvm/prebuilt/darwin-x86_64/lib64/clang/14.0.6/lib/linux/libclang_rt.builtins-arm-android.a(aeabi_memset.S.o): symbol __aeabi_memclr4@@LIBC_N has undefined version LIBC_N
ld: error: /Users/mindgood/Library/Android/sdk/ndk/25.1.8937393/toolchains/llvm/prebuilt/darwin-x86_64/lib64/clang/14.0.6/lib/linux/libclang_rt.builtins-arm-android.a(aeabi_memmove.S.o): symbol __aeabi_memmove4@@LIBC_N has undefined version LIBC_N
ld: error: /Users/mindgood/Library/Android/sdk/ndk/25.1.8937393/toolchains/llvm/prebuilt/darwin-x86_64/lib64/clang/14.0.6/lib/linux/libclang_rt.builtins-arm-android.a(aeabi_memmove.S.o): symbol __aeabi_memmove8@@LIBC_N has undefined version LIBC_N
ld: error: /Users/mindgood/Library/Android/sdk/ndk/25.1.8937393/toolchains/llvm/prebuilt/darwin-x86_64/lib64/clang/14.0.6/lib/linux/libclang_rt.builtins-arm-android.a(aeabi_memset.S.o): symbol __aeabi_memset4@@LIBC_N has undefined version LIBC_N
ld: error: /Users/mindgood/Library/Android/sdk/ndk/25.1.8937393/toolchains/llvm/prebuilt/darwin-x86_64/lib64/clang/14.0.6/lib/linux/libclang_rt.builtins-arm-android.a(aeabi_memset.S.o): symbol __aeabi_memset8@@LIBC_N has undefined version LIBC_N
clang++: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [libgrpc_csharp_ext.so] Error 1
make[1]: *** [CMakeFiles/grpc_csharp_ext.dir/all] Error 2
make: *** [all] Error 2
mindgood@minds-MacBook-Pro grpc % 
```

So far the problem recurs.

Do you need any more info？

### Affected versions

r25

### Canary version

NDK verison is 25.1.8937393

### Host OS

Mac

### Host OS version

macOS Monterey  veriosn 12.6

### Affected ABIs

armeabi-v7a

### Build system

ndk-build

### Other build system

Android SDK Build-Tools verison is 33

### minSdkVersion

Android SDK Build-Tools verison is 33

### Device API level

I'm building static library of grpc

**Comment by DanAlbert** [comment]
> Do you need any more info?

Yes, the bug template asked for repro steps.

**Comment by mkzpd** [comment]
> > Do you need any more info?
> 
> Yes, the bug template asked for repro steps. You've given a CMake command but no source.

@DanAlbert , hi, I just updated my question to be more detailed.

**Comment by DanAlbert** [comment]
Did you find a solution?

**Comment by mkzpd** [comment]
> Did you find a solution?

No,I think using a dynamic link library might be a better option.
See below:
https://chromium.googlesource.com/external/github.com/grpc/grpc/+/HEAD/BUILDING.md#building-with-cmake

**Comment by DanAlbert** [comment]
Going to reopen this then. It may still be a bug.

**Comment by DanAlbert** [comment]
(but we'll treat it as lower priority since it _looks_ like just a configuration error)

**Comment by DanAlbert** [comment]
> brew install openssl

That installs mac binaries, not android. You need to use an openssl built for Android. https://maven.google.com/web/index.html?q=openssl#com.android.ndk.thirdparty:openssl
